processor/src/main/java/org/mapstruct/ap/internal/model/LifecycleCallbackMethodReference.java

```diff
@@ -37,19 +38,70 @@
 
     private final Type declaringType;
     private final List<Parameter> parameterAssignments;
+    private final Type methodReturnType;
+    private final Type methodResultType;
+    private final String instanceVariableName;
+    private final String targetVariableName;
 
-    public LifecycleCallbackMethodReference(SourceMethod method, List<Parameter> parameterAssignments) {
+    public LifecycleCallbackMethodReference(SourceMethod method, List<Parameter> parameterAssignments,
+                                            Type methodReturnType, Type methodResultType,
+                                            Set<String> existingVariableNames) {
         super( method );
         this.declaringType = method.getDeclaringMapper();
         this.parameterAssignments = parameterAssignments;
+        this.methodReturnType = methodReturnType;
```

Comment:
I think it's the return type of the mapping method from which this lifecycle method is called, right?